File: Data Structures & Algorithms/binary-search/submission-5.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1

        while l <= r:
            mid_point = (l + r) // 2

            if target < nums[mid_point]:
                r = mid_point - 1
            elif target > nums[mid_point]:
                l = mid_point + 1
            else:
                return mid_point

        return -1

nums=[8]
solution = Solution()
print(solution.search(nums, 8))

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-5.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] res = new int[nums.Length];
        List<int?> ints = new List<int?>();

        for (int i = 0; i < nums.Length; i++)
        {
            int? prod = null;
            // access all index's but it's current one
            for(int j = 0; j < nums.Length; j++)
            {
                if (j != i)
                {
                    if(prod is not null)
                    {
                        prod *= nums[j];
                    }
                    else
                    {
                        prod = nums[j];
                    }
                }
            }
            ints.Add(prod);
        }

        for(int i = 0; i < nums.Length; i++)
        {
            res[i] = (int)ints[i];
        }

        return res;
    }
}
